### PR TITLE
chore: update template examples and docs to install code-server under /tmp

### PR DESCRIPTION
**Pull request by johnstcn**
This PR updates the example templates to install code-server under /tmp using the standalone method, which should require no root privileges.

We found that the default Kubernetes template in OpenShift would fail to install code-server due to lack of sudo access.

### CODE REVIEW

File: examples/templates/aws-ecs-container/main.tf [hunk 1]
```diff
@@ -102,9 +102,10 @@ resource "coder_agent" "coder" {
   startup_script         = <<-EOT
     set -e
 
-    # install and start code-server
-    curl -fsSL https://code-server.dev/install.sh | sh -s -- --version 4.8.3
-    code-server --auth none --port 13337 >/tmp/code-server.log 2>&1 &
```
Comment (mafredri) [inline examples/templates/aws-ecs-container/main.tf:107]: Did we want to remove logging from this template (same for the others)?

File: examples/templates/aws-ecs-container/main.tf [hunk 2]
```diff
@@ -102,9 +102,10 @@ resource "coder_agent" "coder" {
   startup_script         = <<-EOT
     set -e
 
-    # install and start code-server
-    curl -fsSL https://code-server.dev/install.sh | sh -s -- --version 4.8.3
```
Comment (mafredri) [inline examples/templates/aws-ecs-container/main.tf:106]: Was dropping `--version` intentional?

File: docs/templates.md [hunk 3]
```diff
@@ -160,12 +160,13 @@ resource "coder_agent" "coder" {
   startup_script = <<EOT
 #!/bin/bash
 
-# install code-server 4.8.3
-curl -fsSL https://code-server.dev/install.sh | sh -s -- --version 4.8.3
+# install code-server 4.8.3 under /tmp/code-server
+curl -fsSL https://code-server.dev/install.sh | sh -s -- --method=standalone --prefix=/tmp/code-server --version 4.8.3
+/tmp/code-server/bin/code-server --auth none --port 13337 &
```
Comment (mafredri) [inline docs/templates.md:165]: This doc didn't previously exec the command here (see below)

File: docs/templates.md [hunk 4]
```diff
@@ -160,12 +160,13 @@ resource "coder_agent" "coder" {
...
 
 # The & prevents the startup_script from blocking so the
 # next commands can run.
-code-server --auth none --port &
+code-server --auth none --port 13337 &
```
Comment (mafredri) [inline docs/templates.md:169]: Missing absolute path.

**Review by mafredri — COMMENTED** [review]
Two (potential) issues with storing in `/tmp`:
1. It might be mounted `tmpfs` in which case it'll use up RAM
2. Could be mounted `noexec` which is a somewhat common security measure

Edit: Only meant this as a "to keep in mind", don't see it as a real blocker.

File: examples/templates/aws-ecs-container/main.tf (hunk 2 above)
Comment (johnstcn) [reply examples/templates/aws-ecs-container/main.tf:106]: No, I'll add it back for now. Dropping `--version` can be a separate PR.

File: examples/templates/aws-ecs-container/main.tf (hunk 1 above)
Comment (johnstcn) [reply examples/templates/aws-ecs-container/main.tf:107]: Yet another oversight on my part 😅

**Review by mafredri — APPROVED** [review]
👌🏻

**Pull request merged**